apps/client/src/features/viewers/lower-thirds/LowerThird.tsx

```diff
@@ -0,0 +1,236 @@
+import { useEffect, useState } from 'react';
+import { useSearchParams } from 'react-router-dom';
+import { Message, OntimeEvent, ViewSettings } from 'ontime-types';
+
+import { overrideStylesURL } from '../../../common/api/apiConstants';
+import NavigationMenu from '../../../common/components/navigation-menu/NavigationMenu';
+import { LOWER_THIRD_OPTIONS } from '../../../common/components/view-params-editor/constants';
+import ViewParamsEditor from '../../../common/components/view-params-editor/ViewParamsEditor';
+import { useRuntimeStylesheet } from '../../../common/hooks/useRuntimeStylesheet';
+import { cx } from '../../../common/utils/styleUtils';
+
+import './LowerThird.scss';
+
+enum srcKeys {
+  title = 'title',
+  subtitle = 'subtitle',
+  presenter = 'presenter',
+  lowerMsg = 'lowerMsg',
+}
+
+enum triggerType {
+  event = 'event',
+  manual = 'manual',
+}
+
+type LowerOptions = {
+  width: number;
+  trigger: triggerType;
+  topSrc: srcKeys;
+  bottomSrc: srcKeys;
+  topColour: string;
+  bottomColour: string;
+  topBg: string;
+  bottomBg: string;
+  topSize: string;
+  bottomSize: string;
+  transition: number;
+  delay: number;
+  key: string;
+  lineColour: string;
+  lineHeight: string;
+};
+
+interface LowerProps {
+  eventNow: OntimeEvent | null;
+  viewSettings: ViewSettings;
+  lower: Message;
+}
+
+export default function LowerThird(props: LowerProps) {
+  const { eventNow, lower, viewSettings } = props;
+  const [searchParams] = useSearchParams();
+  const options: LowerOptions = {
+    width: 45,
+    trigger: triggerType.event,
+    topSrc: srcKeys.title,
+    bottomSrc: srcKeys.subtitle,
+    topColour: '000000ff',
+    bottomColour: '000000ff',
+    topBg: '00000000',
+    bottomBg: '00000000',
+    topSize: '65px',
+    bottomSize: '40px',
+    transition: 3,
+    delay: 3,
+    key: 'ffffffff',
+    lineColour: 'ff0000ff',
+    lineHeight: '0.4em',
+  };
```

Comment:
for the `options` object, I'm seeing that you are declaring it then making changes to it with the if statements below according to `searchParams` right?

Problem is every time the component re-renders, it'll redo all the logic - from declaring the options to making the changes)

one thing you could do is wrap this on a `useMemo` hook, `useMemo` will allow you to return the options object with all of the modifications you need done, and it'll only run when the properties in the dependency array change, so for example:

you'd have a 
```
const defaultOptions = { 
 width: 45,
 ....
}
```

and then you could do

```
const options = useMemo(() => {
  const newOptions = {...defaultOptions}
  ...
  make all the changes with the if statements
  ...
  return newOptions;
}, [searchParams])
```

Then it'll only recalculate stuff if the `searchParams` change